Library/Homebrew/utils/analytics.rb

```diff
@@ -16,21 +16,26 @@ def report(type, metadata = {})
         args = %W[
           --max-time 3
           --user-agent #{HOMEBREW_USER_AGENT_CURL}
-          -d v=1
-          -d tid=#{ENV["HOMEBREW_ANALYTICS_ID"]}
-          -d cid=#{ENV["HOMEBREW_ANALYTICS_USER_UUID"]}
-          -d aip=1
-          -d an=#{HOMEBREW_PRODUCT}
-          -d av=#{HOMEBREW_VERSION}
-          -d t=#{type}
+          --data v=1
+          --data aip=1
+          --data-urlencode t=#{type}
+          --data-urlencode tid=#{ENV["HOMEBREW_ANALYTICS_ID"]}
+          --data-urlencode cid=#{ENV["HOMEBREW_ANALYTICS_USER_UUID"]}
+          --data-urlencode an=#{HOMEBREW_PRODUCT}
+          --data-urlencode av=#{HOMEBREW_VERSION}
         ]
-        metadata.each { |k, v| args << "-d" << "#{k}=#{v}" if k && v }
+        metadata.each do |key, value|
+          next unless key
+          next unless value
+          args << "--data-urlencode" << "#{key}=#{value}"
```

Comment:
According to the `curl(1)` man page, the name (the thing before the `=`) is expected to be already URL-encoded. If you wanted to be on the safe side, it would make sense to simply reject keys that are too fancy (I think anything that doesn't match `/^\w+$/` falls into that category) or alternatively URL-encode them on the Ruby side.